components/list/list.js

```diff
@@ -34,13 +36,22 @@ class List extends LitElement {
 	}
 
 	render() {
+		const role = !this.grid ? 'list' : undefined;
 		return html`
-			<div role="list" class="d2l-list-container">
+			<div role="${ifDefined(role)}" class="d2l-list-container">
 				<slot></slot>
 			</div>
 		`;
 	}
 
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('grid')) {
+			const items = this._getItems();
+			items.forEach(item => item.role = this.grid ? 'rowgroup' : 'listitem');
```

Comment:
It's not super amazing for the `d2l-list` to reach down to its children and mutate them.  We do do this sort of thing every once in a while though, but it always feels wrong.  

Currently this is tied into the update for the `grid` property of the list, which won't catch cases where additional items are added to the list (ex. load more), or if items are transplanted from one list to another.  A better place to capture those scenarios is in a slotchange` handler for the slot.

An alternative approach (that we've also utilized) is to have the item check it's closest `d2l-list` ancestor when it's connected. It is possible to have timing/race issues though, and unless the `grid` attribute of the list os observed, it won't be notified of changes... so also not amazing.

Reply:
Yeah, I agree this felt weird. Your point about an update to the list items not occurring when transplanted is important - I think using a `slotchange` handler is the best option for this, partly to avoid having to observe changes to the ancestor `d2l-list` from the list item.

@m6jones What do you think?

Reply:
Slot change would be the best. 

Another approach which might be better is I would expect this value not to change. Ie, you make an aria-grid navigation or list navigation. The application would be super confusing if we switched it up it locations. UX would be terrible.

So doing the alternative approach that Dave suggested is good since the problem shouldn't happen. We could enforce it by having d2l-list and d2l-list-aria-grid components (better names pending)

Reply:
I'll change this to be controlled from the `list-item-generic-layout` checking its ancestor on connection